ecosystem/sep-0008.md

```diff
@@ -61,107 +72,211 @@ approval_server="https://goat.io/tx_approve"
+`message` | string | A human readable string containing information to pass on to the user (optional).
 
-#### Pending Response
+###### Example
 
-A Pending response will have a `200` HTTP status code and `pending` as the `status` value. It means that the issuer needs to asynchronously validate the transaction. The user should resubmit the transaction after a given timeout.
+```json
+{
+  "status": "success",
+  "tx": "AAAAAHAHhQtYBh5F2zA6..."
+}
+```
+
+##### Revised
+
+This response means that the transaction was revised to be made compliant.
+
+A `revised` response will have a `200` HTTP status code and `revised` as the `status` value.
+
+It is important for the user to examine the revised transaction before re-signing it. More specifically, the user should make sure that the revised transaction does not contain any additional operations whose source account matches the source account of the original operation(s).
 
 Parameters:
 
-Name | Data Type | Description
------|-----------|------------
-status|string|`"pending"`
-timeout|integer|Number of milliseconds to wait before submitting the same transaction again.
-message|string|A human readable string containing information to pass on to the user (optional).
+Name | Type | Description
+-----|------|------------
+`status` | string | `"revised"`
+`tx` | string | Transaction envelope XDR, base64 encoded. This transaction is a revised compliant version of the original request transation, signed by the issuer.
+`message` | string | A human readable string explaining the modifications made to the transaction to make it compliant.
+
+###### Example
+
+```json
+{
+  "status": "revised",
+  "tx": "AAAAAHAHhQtYBh5F2zA6...",
+  "message": "Authorization and deauthorization operations were added."
+}
+```
 
-#### Action Required Response
+##### Pending
 
-An Action Required response will have a `200` HTTP status code and `action_required` as the `status` value. It means that the user must complete an action before this transaction can be approved. The approval service will provide a URL that facilitates the action. Upon completion, the user will resubmit the transaction.
+This response means that the issuer could not determine whether to approve the transaction at the time of receiving it. Wallet can re-submit the same transaction at a later point in time.
+
+A `pending` response will have a `200` HTTP status code and `pending` as the `status` value.
 
 Parameters:
 
-Name | Data Type | Description
------|-----------|------------
-status|string|`"action_required"`
-message|string|A human readable string containing information regarding the action required.
-action_url|string|A URL that allows the user to complete the actions required to have the transaction approved.
+Name | Type | Description
+-----|------|------------
+`status` | string | `"pending"`
+`timeout` | integer | Number of milliseconds to wait before submitting the same transaction again (optional).
```

Comment:
>This PR adds three non-breaking changes/improvement to SEP-8

❗ Making this field optional is a breaking change. Would it be sufficient for a server to return zero if there is no timeout period required? That would require no change to the SEP.

Reply:
Ah you're right. If a client checks for the field, it will result in a failure if it is missing. I've removed this change.

> Would it be sufficient for a server to return zero if there is no timeout period required? That would require no change to the SEP.

It could be, but I think we need to explain what a zero timeout means. I've done it in [this](https://github.com/stellar/stellar-protocol/pull/818) PR.